plugins/chat/app/services/chat/auto_join_channels.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Chat
+  class AutoJoinChannels
+    include Service::Base
+
+    ALLOWED_GROUP_PERMISSIONS = [
+      CategoryGroup.permission_types[:create_post],
+      CategoryGroup.permission_types[:full],
+    ]
+
+    policy :chat_enabled?
```

Comment:
I kind of agree that we should have at least a basic level of test coverage for these service classes. For example turning off chat should `fail_a_policy(:chat_enabled)`